src/compositemark/boxplot.ts

```diff
@@ -31,17 +36,62 @@ export function normalizeBoxPlot(spec: GenericUnitSpec<Encoding<Field>, BOXPLOT>
       discreteAxis = 'x';
       continuousAxis = 'y';
       continuousAxisChannelDef = encoding.y;
-
       discreteAxisFieldDef = encoding.x;
     } else if (isDiscrete(encoding.y) && isContinuous(encoding.x)) {
       // horizontal
       discreteAxis = 'y';
       continuousAxis = 'x';
       continuousAxisChannelDef = encoding.x;
-
       discreteAxisFieldDef = encoding.y;
     } else {
-      throw new Error('Need one continuous and one discrete axis for 2D boxplots');
+      if (isContinuous(encoding.x) && isContinuous(encoding.y)) {
```

Comment:
This chunk of logic looks very lengthy.   

I think you can have one method to determine the orientation and another to determine `discreteAxis`, `continuousAxis`, `continuousAxisChannelDef` and `discreteAxisFieldDef`.  

Something like: 

```
const orient = boxOrient(...);
const {discreteAxisFieldDef, continuousAxis, continuousAxisChannelDef, discreteAxisFieldDef} = boxParams(<...>, orient);
```